src/EmbedIndex/Program.cs

```diff
@@ -164,33 +165,85 @@ public static IEnumerable<string> ComputeIndexEntries(string archiveRelativePath
+                    ZipArchiveEntry catalogIndexEntry = archive.CreateEntry(CatalogIndexFileName);
+                    using (Stream indexStream = catalogIndexEntry.Open())
+                    {
+                        using (TextWriter writer = new StreamWriter(indexStream))
+                        {
+                            writer.WriteLine(catalogHash);
+                            writer.Flush();
+                        }
+                    }
                 }
-                string signedFile = catalogEntry.FullName.Substring(0, catalogEntry.FullName.Length - 4);
-                ZipArchiveEntry signedFileEntry = archive.Entries.SingleOrDefault(e => e.FullName == signedFile);
-                if (signedFileEntry == null)
+            }
+
+            // get list of cataloged symbol files
+            ZipArchiveEntry catalogedFilesEntry = archive.GetEntry(CatalogedSymbolListFileName);
+
+            if (catalogedFilesEntry == null)
+            {
+                yield break;
+            }
+
+            string[] catalogedFiles;
+            using (Stream catalogedFilesStream = catalogedFilesEntry.Open())
+            {
+                using (StreamReader reader = new StreamReader(catalogedFilesStream))
                 {
-                    continue;
+                    string content = reader.ReadToEnd();
+                    catalogedFiles = content.Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries);
                 }
+            }
+            catalogedFilesEntry.Delete();
+
+            // for each regular symbol index entry covered by a symbol catalog,
+            // add an entry for the catalog file
+            HashSet<Tuple<string, string>> seen = new HashSet<Tuple<string, string>>();
+            foreach (string catalogedFile in catalogedFiles)
+            {
+                ZipArchiveEntry fileEntry = archive.GetEntry(catalogedFile);
                 foreach (IFileFormatIndexer indexer in indexers)
                 {
-                    using (Stream signedFileStream = signedFileEntry.Open())
+                    using (Stream signedFileStream = fileEntry.Open())
                     {
-                        string key = indexer.ComputeIndexKey(signedFile, signedFileStream);
+                        string key = indexer.ComputeIndexKey(fileEntry.FullName, signedFileStream);
                         if (key != null)
                         {
-                            key = key.Substring(0, key.LastIndexOf("/") + 1) + catalogEntry.Name.ToLowerInvariant();
-                            yield return Tuple.Create(key, catalogEntry.FullName);
+                            key = key.Substring(0, key.LastIndexOf("/") + 1) + CatalogIndexFileName.ToLowerInvariant();
+                            Tuple<string, string> ret = Tuple.Create(key, CatalogIndexFileName);
+                            if (!seen.Contains(ret))
+                            {
+                                seen.Add(ret);
```

Comment:
Changed as suggested.  I saw this in the Microsoft.CSharp package - the netcore50 and netstandard1.3 versions of Microsoft.CSharp.dll are indentical so without this we had two identical entries (since our catalog entries don't include the path the way symbol entries do).